### ISSUE TITLE
SFTPClient.upload_file is way slower than the others functions

### ISSUE THREAD

**Issue by jipen** [i22-digitalagentur/sftp_client#2]
First, thanks a lot for SFTPClient !
I'm currently [developping a SFTP Client/Server](https://elixirforum.com/t/secure-files-transfers-servers-in-elixir/21119) and after using another SFTP library, I give a try to yours at the moment.
I made some tests on a local Windows PC (2 SFTP servers on differents ports exchanging files).
The transfer rates are all very nice (from 23 to 41 Mb/s), streamed or not...
Except for direct upload (SFTPClient.upload_file function)  for which the rate falls down to 0,6 to 0.9 Mb/s only.
I isolated yours functions (SFTPClient.connect and SFTPClient.upload_file) in case this fall could come from my own code but with the same results.
Do you have an idea about what could cause that ?
Best regards.

**Comment by tlux** [comment]
Hello @jipen and thank you for your bug report.

I'm quite sure it has something to do with the chunk size that is currently used (or better NOT used) in the `UploadFile` module. While `DownloadFile` transmits data in chunks of 32768 Bytes, `UploadFile` transmits each line of a file individually. This is probably the reason why it it so slow at the moment. 

However, I created a branch in which I incremented the chunk size for uploads:
https://github.com/i22-digitalagentur/sftp_client/tree/increase-upload-chunk-size

You're invited to try it out and post your feedback here. :)  

```elixir
def deps do
  [
    {:sftp_client, 
     github: "i22-digitalagentur/sftp_client", 
     branch: "increase-upload-chunk-size"}
  ]
end
```

**Comment by jipen** [comment]
Hi Tlux,
BTW I didn't manage to see that you don't use the Erlang ssh_sftp functions (such as write_file or read_file) to implement your upload_file or download_file functions as it is done in Sftp_Ex.
Now I've seen: Streaming all the way ! ;)
Indeed, considering my project's constraints (asynchronous big files transfers) I also onky use streaming functions to avoid memory overloading.
I was just benchmarking and giving a try to your upload_file and download_file functions when I noticed this upload_file slow rate. 
So I only wanted to inform you about this problem.
I'm sure that you have found the solution with this chunk size as none of my others tests show this rate problem.
Anyway, thanks again for your work ! 
PS: If I find some time tomorrow, I'll test your branch.

**Comment by jipen** [comment]
upload_file function is OK now.

**Comment by tlux** [comment]
The fix is now published in version 1.3.4. Thanks again for the hint!
